### PR TITLE
Use wharf-core problem.Response instead

### PR DESCRIPTION
**Pull request by jilleJr**
Using new unified problem.Response from iver-wharf/wharf-core#9 instead of a copy-pasted one from wharf-api.

### CODE REVIEW

File: pkg/wharfapi/problem.go [hunk 1]
```diff
@@ -1,61 +1,51 @@
...
+		return problem.Response{}, fmt.Errorf(
+			"failed to close problem response body reading: %w", closeErr)
+	}
+	var prob problem.Response
+	if jsonErr := json.Unmarshal(resp, &prob); jsonErr != nil {
+		return problem.Response{}, fmt.Errorf(
+			`failed to parse "%s" problem response: %w`, problemContentType, jsonErr)
+	}
+	return prob, nil
+}
```
Comment (Pikabanga) [inline pkg/wharfapi/problem.go:43]: Maybe some of these could be moved to core?

**Review by Pikabanga — APPROVED** [review]

File: pkg/wharfapi/problem.go (hunk 1 above)
Comment (jilleJr) [reply pkg/wharfapi/problem.go:43]: Good idea!

Comment (jilleJr) [reply pkg/wharfapi/problem.go:43]: https://github.com/iver-wharf/wharf-core/pull/12

Comment (jilleJr) [reply pkg/wharfapi/problem.go:43]: Will change this repo in a new PR once that wharf-core one has been merged.

**Review by Alexamakans — APPROVED** [review]

**Pull request merged**